### ISSUE TITLE
Can not save features - not enough points

### ISSUE THREAD

**Issue by islam500** [nextgis/android_gisapp#569]
The current code base compiles but does not allow saving of new features or feature updates. It says "Not enough Points"

**Comment by BishopGIS** [comment]
That is geometry type of layer?
That was datadource? NextGIS.com, geojson, manually created. If it was a file source can you provide test data?
That are reproduced steps?

**Comment by islam500** [comment]
Hello Dmitry. The problem is seen in all types of layers (point, polygon, and line) and all sources of data (Nextgis, local, default layers).

I suspect the issue might be from the maplib and maplibui libraries. because when I checked out the project those folders were empty. and I noticed even in the zipped release codes for both 2.4 and 2.5 those folders are empty. I had to download them directly from the repository and extract them to my local setup. Its probably a version miss match.

![maplib](https://cloud.githubusercontent.com/assets/2852014/26569237/4ce8d6d8-4501-11e7-8754-ec8db805974a.JPG)

**Comment by BishopGIS** [comment]
**Maplib** and **maplibui** are mandatory to build NextGIS Mobile application.
You have to clone repository with submodules using this command:

> git clone --recursive https://github.com/nextgis/android_gisapp.git

Also you can init submodules in already clonned repository 

> git submodule update --init --recursive

Usually Android Studio clone with submodules via interface.

![2017-05-30 11 37 10](https://cloud.githubusercontent.com/assets/1708122/26575062/65776202-452c-11e7-9fe8-afb832ab3abf.png)

Is this solution suits you?

**Comment by islam500** [comment]
Thank you Dmitry for that. I have used the git command with recursive to check out the complete repository including the maplib and maplibui this time. But the problem still persists in both emulator, Samsung Galaxy S7edge phone and Samsung tab tablet.

However the problem is not present when building the 2.5 release.

I have attached an emulator screen shot. 
![not_enough_points](https://cloud.githubusercontent.com/assets/2852014/26584115/9ecaf7b8-453f-11e7-8913-b22ba681e507.JPG)
